crates/hir/src/semantics.rs

```diff
@@ -1236,6 +1236,15 @@ impl<'db> SemanticsImpl<'db> {
         sa.resolve_macro_call(self.db, macro_call)
     }
 
+    pub fn is_proc_macro_call(&self, macro_call: &ast::MacroCall) -> bool {
+        let sa = match self.analyze(macro_call.syntax()) {
+            Some(it) => it,
+            None => return false,
+        };
+        let macro_call = self.find_file(macro_call.syntax()).with_value(macro_call);
+        sa.is_proc_macro_call(self.db, macro_call)
```

Comment:
```suggestion
        self.resolve_macro_call(macro_call).map_or(false, |m| matches!(m.id, MacroId::ProcMacroId(..))
```

No need for `SourceAnalyzer::is_proc_macro_call` then